frontend/components/cards/ConversationCard.tsx

```diff
@@ -1,34 +1,32 @@
-import React, { useMemo } from 'react';
+import React from 'react';
 import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
 import { theme } from '../../styles/theme';
-import { IConversation, IMessage } from '../../types/databaseTypes';
+import { IConversation } from '../../types/databaseTypes';
 import { useQuery } from '@tanstack/react-query';
 import pictureStyles from '../../styles/pictureStyles';
 import defaultStyles from '../../styles/defaultStyles';
 import { Circle, Svg } from 'react-native-svg';
 import { getDateText } from '../../utils/dateToText';
 import useBarkNavigation from '../../hooks/useBarkNavigation';
 import { getUserByID } from '../../api/user/getUserByID';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../redux/store';
 
 interface ConversationCardProps {
   conversation: IConversation;
 }
 
 const ConversationCard = ({ conversation }: ConversationCardProps) => {
   const navigation = useBarkNavigation();
+  const userId = useSelector((state: RootState) => state.auth.userData);
+  const hasNewMessage = !conversation.messages[0].read.includes(userId);
```

Comment:
I think this will crash if messages is an empty array. Should probably be:

```suggestion
  const hasNewMessage = !conversation.messages[0]?.read.includes(userId);
```

Reply:
Nice spotted!
![bitmoji](https://sdk.bitmoji.com/render/panel/43533bc8-de9f-4e89-9f7d-f9a64b6254c4-eb1f676a-1fcc-49de-809c-bcc58ba85a75-v1.png?transparent=1&palette=1&width=246)